packages/opentelemetry-node/src/NodeTracer.ts

```diff
@@ -0,0 +1,71 @@
+/*!
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*!
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as api from '@opentelemetry/api';
+import { setActiveSpan } from '@opentelemetry/core';
+import {
+  BasicTracerProvider,
+  Tracer,
+  TracerConfig,
+} from '@opentelemetry/tracing';
+import { AsyncHooksContextManager } from '@opentelemetry/context-async-hooks';
+
+/**
+ * This class represents a nodejs-specific tracer.
+ */
+export class NodeTracer extends Tracer {
+  /**
+   * Constructs a new NodeTracer instance.
+   */
+  constructor(config: TracerConfig, _tracerProvider: BasicTracerProvider) {
+    super(config, _tracerProvider);
+  }
+
+  async withSpanAsync<
+    T extends Promise<any>,
+    U extends (...args: unknown[]) => T
+  >(span: api.Span, fn: U): Promise<T> {
+    // @ts-ignore
```

Comment:
see PR's description

Reply:
I can think of a couple solutions that do not involve ts-ignore

1. Expose `_getContextManager` publicly as `getContextManager`. Some of the other apis (trace at least) already do this
2. Create a `withContextAsync` method on the `context` api. This would need to be in some way marked as experimental or unstable. Probably something in the ts doc.

Reply:
The problem is that `withContextAsync` cannot be implemented at the API level because it's not possible to make it work in the browser. Or we need to make it a noop on the browser ? WDYT ?

Reply:
ref to previous discussion: https://github.com/open-telemetry/opentelemetry-js/pull/926#discussion_r405591287

Reply:
Option 1 still doesn't have this problem

Reply:
Sure, i'll to do that then